### DISCUSSION TITLE
make subtitles shorter!

### DISCUSSION THREAD

**Discussion by GilorHatahat** [openai/whisper#1672] [Ideas]
for the love of god make us able to pick values higher than 1 on length_penalty. 
the subtitles outputed are way too long and I can't find any way to make whisper cut them shorter, like whenever theres a period in the speech.
does anyone have a solution?

**Comment by glangford** [comment]
From `whisper --help`- there are options for customizing subtitles:

```
 --max_line_width MAX_LINE_WIDTH
                        (requires --word_timestamps True) the maximum number of characters in a line before breaking the line (default: None)
  --max_line_count MAX_LINE_COUNT
                        (requires --word_timestamps True) the maximum number of lines in a segment (default: None)
```

**Reply by GilorHatahat** [reply]
the first one breaks the subtitle into two lines and the second one chooses how many lines can be in one subtitle.
what I want to do is make multiple short subtitles that dont have line breaks.
actually now that i think about it, maybe its possible to put small line width like 42 and then do a line count of 1 so that it will do a bunch of small subtitles instead of the long ones. ill try that.
edit: tried it. doesnt work.

**Reply by glangford** [reply]
try

`whisper <your file> --word_timestamps True --max_line_width 42 --max_line_count 1 --output_format srt`

and review *the .srt file that is produced* to see if that works for you, not the console log.

**Comment by GilorHatahat** [comment]
> From `whisper --help`- there are options for customizing subtitles:
> 
> ```
>  --max_line_width MAX_LINE_WIDTH
>                         (requires --word_timestamps True) the maximum number of characters in a line before breaking the line (default: None)
>   --max_line_count MAX_LINE_COUNT
>                         (requires --word_timestamps True) the maximum number of lines in a segment (default: None)
> ```



> try
> 
> `whisper <your file> --word_timestamps True --max_line_width 42 --max_line_count 1 --output_format srt`
> 
> and review _the .srt file that is produced_ to see if that works for you, not the console log.

didnt work as expected sadly.

**Reply by rohitkrishna094** [reply]
Were you able to figure this out @GilorHatahat ?

**Comment by LunaMyan** [comment]
@GilorHatahat I'm having the same issue. Were you able to work it out?

**Comment by srvanrell** [comment]
`--max_words_per_line N` in combination with  `--word_timestamps True` has worked for me
